guides/hyperlane-use-tia-for-gas.md

```diff
@@ -0,0 +1,437 @@
+# How to use Hyperlane token (TIA) as gas token in your rollup
+
+This guide builds upon [CosmWasm rollup](../tutorials/cosmwasm.md), where you set up a local rollup with integrated CosmWasm. In this guide, we will be exploring how to integrate Hyperlane as a safer IBC alterative for token transfers in your rollup.
+
+<!-- markdownlint-disable MD033 -->
+<script setup>
+import Callout from '../.vitepress/components/callout.vue'
+</script>
+
+:::tip
+<Callout />
+:::
+
+<!-- markdownlint-enable MD033 -->
+
+## 📝 Deploy a CosmWasm rollup {#deploy-rollup}
+
+This is a gist of the [CosmWasm rollup](../tutorials/cosmwasm.md) guide.
+
+### Prepare rollup binary
+
+```bash
+git clone --branch v0.50.0 --depth 1 https://github.com/CosmWasm/wasmd.git
+cd wasmd
+```
+
+#### Add token factory to app.go:
+
+```bash
+curl -sSL https://raw.githubusercontent.com/Stride-Labs/Rollkit-docs/hyperlane-v2/guides/assets/hyperlane-use-tia-for-gas/add-tokenfactory.diff | git apply
```

Comment:
As opposed to blinding applying a git diff, it would be great to understand what the diff is so that people understand it. Similar to how we walk the user through building the wordle tutorial.

Reply:
There are tons of parts like this during the guide, it'll be super long and boring if we explain all of these, it's like going through the commit history and explaining commits. Would you rather we fork the repo, apply the patch and have people clone our fork?

Reply:
Can we instead have a tag or branch that we are targeting? 
A branch would be nice because then it would be easier to know how to update in the future. 

Then the steps would be:
```
git clone https://github.com/Stride-Labs/wasmd
cd wasmd
git checkout rollkit-hyperlane
```

That would be my preferred way.